lighthouse-core/gather/gatherers/dobetterweb/mutation-events.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Tests whether the page is using Date.now().
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+const MUTATION_EVENTS = [
+  'DOMAttrModified',
+  'DOMAttributeNameChanged',
+  'DOMCharacterDataModified',
+  'DOMElementNameChanged',
+  'DOMNodeInserted',
+  'DOMNodeInsertedIntoDocument',
+  'DOMNodeRemoved',
+  'DOMNodeRemovedFromDocument',
+  'DOMSubtreeModified'
+];
+
+class MutationEventUse extends Gatherer {
+
+  beforePass(options) {
+    this.collectUsage = options.driver.captureFunctionCallSites('addEventListener');
+    this.collectUsage2 = options.driver.captureFunctionCallSites('document.addEventListener');
+    // TODO: document.body appears to not be defined by this time.
+    // this.collectUsage3 = options.driver.captureFunctionCallSites('document.body.addEventListener');
```

Comment:
there's also [`DOMDebugger.getEventListeners`](https://chromedevtools.github.io/debugger-protocol-viewer/tot/DOMDebugger/#method-getEventListeners). Would that work? Seems like it would be much more lightweight than this approach.